tests/store/test_obssurface.py

```diff
@@ -1,15 +1,38 @@
 import json
-
 import pytest
 import xarray as xr
 from helpers import attributes_checker_obssurface, get_surface_datapath, clear_test_stores
-from openghg.objectstore import exists, get_bucket
+from openghg.objectstore import exists, get_bucket, get_writable_bucket
 from openghg.store import ObsSurface
 from openghg.store.base import Datasource
 from openghg.util import create_daterange_str
 from pandas import Timestamp
 
 
+def test_raising_error_doesnt_save_to_store(mocker):
+    clear_test_stores()
+    bucket = get_writable_bucket(name="user")
+
+    key = ""
+    with pytest.raises(ValueError):
+        with ObsSurface(bucket=bucket) as obs:
+            key = obs.key()
+            assert not exists(bucket=bucket, key=key)
+            raise ValueError("Oops")
```

Comment:
Yeah, this is pretty much just checking the Python implementation so might not be best way but I just wanted to make sure the way I setup the handling of errors worked as I expected.

Reply:
So is this checking that raising a `ValueError` from in the `with` statement is doing the right thing?

Reply:
Yeah, in case people do something else with `obs` that raises an error. Then we test what happens if `assign_data` raises an error below.

Reply:
I think a quick comment here to that effect would help make sure we're clear on what this is checking for. I imagine we could get a bit confused otherwise.